components/site/src/link_checking.rs

```diff
@@ -3,84 +3,68 @@ use rayon::prelude::*;
+            p.internal_links.iter().map(move |l| (path.clone(), l))
         })
         .flatten();
-    let all_links = page_links.chain(section_links).collect::<Vec<_>>();
-
-    if site.config.is_in_check_mode() {
-        println!("Checking {} internal link(s) with an anchor.", all_links.len());
-    }
-
-    if all_links.is_empty() {
-        return Ok(());
-    }
-
-    let mut full_path = site.base_path.clone();
-    full_path.push("content");
-
-    let errors: Vec<_> = all_links
-        .iter()
-        .filter_map(|(page_path, (md_path, anchor))| {
-            // There are a few `expect` here since the presence of the .md file will
-            // already have been checked in the markdown rendering
-            let mut p = full_path.clone();
-            for part in md_path.split('/') {
-                p.push(part);
-            }
-            if md_path.contains("_index.md") {
-                let section = library
-                    .get_section(&p)
-                    .expect("Couldn't find section in check_internal_links_with_anchors");
-                if section.has_anchor(&anchor) {
-                    None
-                } else {
-                    Some((page_path, md_path, anchor))
-                }
-            } else {
-                let page = library
-                    .get_page(&p)
-                    .expect("Couldn't find section in check_internal_links_with_anchors");
-                if page.has_anchor(&anchor) {
-                    None
-                } else {
-                    Some((page_path, md_path, anchor))
-                }
-            }
+    let all_links = page_links.chain(section_links);
+
+    // Only keep links with anchor fragments, and count them too.
+    // Bare files have already been checked elsewhere, thus they are not interesting here.
+    let mut anchors_total = 0usize;
+    let links_with_anchors = all_links
+        .filter_map(|(page_path, link)| match link {
+            (md_path, Some(anchor)) => Some((page_path, md_path, anchor)),
+            _ => None,
         })
-        .collect();
-
-    if site.config.is_in_check_mode() {
-        println!(
-            "> Checked {} internal link(s) with an anchor: {} error(s) found.",
-            all_links.len(),
-            errors.len()
-        );
-    }
-
-    if errors.is_empty() {
-        return Ok(());
-    }
+        .inspect(|_| anchors_total = anchors_total.saturating_add(1));
+
+    // Check for targets existence (including anchors), then keep only the faulty
+    // entries for error reporting purposes.
+    let missing_targets = links_with_anchors.filter(|(_, md_path, anchor)| {
+        // There are a few `expect` here since the presence of the .md file will
+        // already have been checked in the markdown rendering
+        let mut full_path = site.base_path.clone();
+        full_path.push("content");
+        for part in md_path.split('/') {
```

Comment:
is that causing issues on Windows?

Reply:
Likely not? This is not newly added logic (see L45 of existing codebase), so if it didn't get any bug reports before I assume it does work on Windows too (but I don't have a machine to test it manually).

To the best of my understanding, Zola does not require internal links to be written differently according to the platform (i.e. `@/foo/bar.md` on Mac, and `@/foo\bar.md` on Windows). Thus I guess `md_path.split('/')` would work everywhere.

Reply:
Could [Path iter](https://doc.rust-lang.org/std/path/struct.Path.html#method.iter) be used here?

Reply:
As far as I can tell this isn't a filesystem path (i.e. the separator does not change according to the runtime OS), so `Path` and its components splitting/iterating logic won't be the correct one to use here.